src/nvim/ex_docmd.c

```diff
@@ -9624,21 +9624,41 @@ static void ex_folddo(exarg_T *eap)
 
 static void ex_terminal(exarg_T *eap)
 {
+  // We will call termopen() with shell as an arglist if not given a {cmd}.
   char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
+  char ex_cmd[1024] = "";
   bool mustfree = false;
-  char *lquote = "['";
-  char *rquote = "']";
 
   if (*eap->arg != NUL) {
     name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
     mustfree = true;
-    lquote = rquote = "\"";
+    snprintf(ex_cmd, sizeof(ex_cmd),
+             ":enew%s | call termopen(\"%s\") | startinsert",
```

Comment:
Turns out that these need to be double quotes, there is a test that `:term` allows quotes and slashes that fails if I change this.